### PR TITLE
Non-alert queries

### PR DESCRIPTION
**Pull request by sauyon**
I was being rather silly; Go gives very good errors about what exactly has gone wrong, and the cases where errors don't have locations are probably much more rare than I thought.

EDIT: More tests on the way, but I'm reworking the test script, so that might take a bit more time.

### CODE REVIEW

File: ql/test/query-tests/Diagnostics/ExtractionErrors.expected [hunk 1]
```diff
@@ -0,0 +1 @@
+ERROR: Diagnostic must implement toString() (/home/sauyon/devel/semmlego/ql/src/Diagnostics/DiagnosticsReporting.qll:6,15-25)
```
Comment (smowton) [inline ql/test/query-tests/Diagnostics/ExtractionErrors.expected:1]: This doesn't look expected?

**Review by smowton — APPROVED** [review]

**Review by smowton — APPROVED** [review]

**Comment by smowton** [comment]
Rebased and set to automerge

**Comment by smowton** [comment]
@sauyon are the test changes as you'd expect?

**Comment by sauyon** [comment]
Yep, let me rebase again...

**Comment by sauyon** [comment]
Had to evaluate symlinks for the other PR; this should hopefully work...

**Pull request merged**